lib/raygun.ts

```diff
@@ -11,44 +9,28 @@
 "use strict";
 
 import type {
-  RawUserData,
-  OfflineStorageOptions,
-  Tag,
   CustomData,
-  RequestParams,
+  Hook,
   Message,
+  IOfflineStorage,
+  OfflineStorageOptions,
+  RawUserData,
+  RaygunOptions,
+  RequestParams,
+  Tag,
+  Transport,
 } from "./types";
 import type { Request, Response, NextFunction } from "express";
 import * as raygunTransport from "./raygun.transport";
 import { RaygunMessageBuilder } from "./raygun.messageBuilder";
 import { OfflineStorage } from "./raygun.offline";
+import { RaygunBatchTransport } from "./raygun.batch";
+
+const debug = require("debug")("raygun");
 
 type SendCB = (error: Error | null, items: string[] | undefined) => void;
 
-type Hook<T> = (
-  message: Message,
-  exception: Error | string,
-  customData: CustomData,
-  request?: RequestParams,
-  tags?: Tag[]
-) => T;
-
-type RaygunOptions = {
-  apiKey: string;
-  filters?: string[];
-  host?: string;
-  port?: number;
-  useSSL?: boolean;
-  onBeforeSend?: Hook<Message>;
-  offlineStorage?: OfflineStorage;
-  offlineStorageOptions?: OfflineStorageOptions;
-  isOffline?: boolean;
-  groupingKey?: Hook<string>;
-  tags?: Tag[];
-  useHumanStringForObject?: boolean;
-  reportColumnNumbers?: boolean;
-  innerErrorFieldName?: string;
-};
+const DEFAULT_BATCH_FREQUENCY = 1000; // ms
```

Comment:
Not sure what a good default would be here. Perhaps a bit longer of a period, but needs benchmarking to see how much that impacts heavy load.

Reply:
I think this will actually be okay to start with. It's easy to change both locally and in releases, so we can worry about it later.